backend/routers/grafana.py

```
@@ -0,0 +1,318 @@
+
+class MetricsBatchItem(BaseModel):
+    query: str
+    start: Optional[int] = None
+    end: Optional[int] = None
+    step: str = "60"
+
+
+@router.post("/metrics/batch")
+async def query_metrics_batch(
+    items: list[MetricsBatchItem],
+    _: dict = Depends(require_admin),
```

Ruff (0.15.15)

[warning] Do not perform function call `Depends` in argument defaults; instead, perform the call within the function, or read the default from a module-level singleton variable

(B008)